recipes/szip/all/conanfile.py

```diff
@@ -28,91 +31,74 @@ class SzipConan(ConanFile):
+        get(self, **self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True)
 
     def build(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
-        tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"),
+        apply_conandata_patches(self)
+        replace_in_file(self, os.path.join(self.source_folder, "CMakeLists.txt"),
                               "set (CMAKE_POSITION_INDEPENDENT_CODE ON)", "")
-        cmake = self._configure_cmake()
+        cmake = CMake(self)
+        cmake.configure()
         cmake.build()
 
-    def _configure_cmake(self):
-        if self._cmake:
-            return self._cmake
-        self._cmake = CMake(self)
-        self._cmake.definitions["SZIP_ENABLE_ENCODING"] = self.options.enable_encoding
-        self._cmake.definitions["SZIP_EXTERNALLY_CONFIGURED"] = True
-        self._cmake.definitions["BUILD_TESTING"] = False
-        self._cmake.definitions["SZIP_BUILD_FRAMEWORKS"] = False
-        self._cmake.definitions["SZIP_PACK_MACOSX_FRAMEWORK"] = False
-        self._cmake.definitions["SZIP_ENABLE_LARGE_FILE"] = self.options.enable_large_file
-        if tools.cross_building(self, skip_x64_x86=True) and self.options.enable_large_file:
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["SZIP_ENABLE_ENCODING"] = self.options.enable_encoding
+        tc.variables["SZIP_EXTERNALLY_CONFIGURED"] = True
+        tc.variables["BUILD_TESTING"] = False
+        tc.variables["SZIP_BUILD_FRAMEWORKS"] = False
+        tc.variables["SZIP_PACK_MACOSX_FRAMEWORK"] = False
+        tc.variables["SZIP_ENABLE_LARGE_FILE"] = self.options.enable_large_file
+        if cross_building(self, skip_x64_x86=True) and self.options.enable_large_file:
             # Assume it works, otherwise raise in 'validate' function
-            self._cmake.definitions["TEST_LFS_WORKS_RUN"] = True
-            self._cmake.definitions["TEST_LFS_WORKS_RUN__TRYRUN_OUTPUT"] = True
-        self._cmake.configure(build_folder=self._build_subfolder)
-        return self._cmake
+            tc.variables["TEST_LFS_WORKS_RUN"] = True
+            tc.variables["TEST_LFS_WORKS_RUN__TRYRUN_OUTPUT"] = True
+        # Honor BUILD_SHARED_LIBS from conan_toolchain (see https://github.com/conan-io/conan/issues/11840)
+        tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0077"] = "NEW"
+        tc.generate()
 
     def package(self):
-        self.copy("COPYING", dst="licenses", src=self._source_subfolder)
-        cmake = self._configure_cmake()
+        copy(self, "COPYING", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        cmake = CMake(self)
         cmake.install()
-        # TODO: to remove in conan v2 once cmake_find_package* generators removed
-        self._create_cmake_module_alias_targets(
```

Comment:
Quickly this should be required for the `test_v1_package` 🤔 

You should check the https://github.com/conan-io/conan-center-index/tree/master/docs/package_templates/cmake_package for a reference about the 2.0 migration

Theres extras in here I am not sure about 

***

If you'd like we can do a zoom to finish this up and it will be easier to work instead of the horrible time zone gap

Reply:
Chris was awesome in the zoom, and so now I'm a little bit wiser than before :)